pandora-common/src/gameLogic/assetPreferences/assetPreferencesSubsystem.ts

```diff
@@ -0,0 +1,48 @@
+import { Immutable } from 'immer';
+import { Asset } from '../../assets/asset';
+import { AssetPreferenceResolution, AssetPreferenceType, AssetPreferencesPublic, ResolveAssetPreference } from '../../character/assetPreferences';
+import { CharacterId } from '../../character/characterTypes';
+import { TypedEventEmitter } from '../../event';
+import { GameLogicPermission, IPermissionProvider, PermissionConfigDefault } from '../permissions';
+
+export type AssetPreferencesSubsystemEvents = {
+	dataChanged: void;
+};
+
+export const ASSET_PREFERENCES_PERMISSIONS = {
+	favorite: {
+		visibleName: 'Favorite',
+		defaultPermissions: {
+			allowOthers: true,
+		},
+	},
+	normal: {
+		visibleName: 'Normal',
+		defaultPermissions: {
+			allowOthers: true,
+		},
+	},
+	maybe: {
+		visibleName: 'Maybe',
+		defaultPermissions: {
+			allowOthers: true,
+		},
+	},
+	prevent: null,
+	doNotRender: null,
+} as const satisfies Immutable<Record<AssetPreferenceType, AssetPreferencePermissionConfig | null>>;
+
+export interface AssetPreferencePermissionConfig {
+	visibleName: string;
+	defaultPermissions: PermissionConfigDefault;
+}
+
+export abstract class AssetPreferencesSubsystem extends TypedEventEmitter<AssetPreferencesSubsystemEvents> implements IPermissionProvider {
+	public abstract get currentPreferences(): Immutable<AssetPreferencesPublic>;
+	public abstract getPreferencePermission(preference: AssetPreferenceType): GameLogicPermission | null;
```

Comment:
This feels like a complete duplication of the other Subsystem, but for some reason, this function can return null, the other throws an exception.

What exactly do we gain from this much extra duplicated code, if we just want to split it for the UI, a simple group in the config would have been able to accomplish the same thing with much less code

Reply:
well I tried this but over all it did not make anything better, the duplication should be fine for now, I guess